aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cu

```diff
@@ -0,0 +1,132 @@
+#include <ATen/Context.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/Dispatch.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/cuda/PinnedMemoryAllocator.h>
+#include <ATen/cuda/CUDAApplyUtils.cuh>
+#include <ATen/cuda/detail/IndexUtils.cuh>
+#include <ATen/cuda/CUDASolver.h>
+#include <ATen/cuda/CUDABlas.h>
+#include <ATen/cuda/CUDAEvent.h>
+#include <c10/cuda/CUDAStream.h>
+
+#include <ATen/native/LinearAlgebraUtils.h>
+#include <ATen/native/cuda/MiscUtils.h>
+#include <ATen/native/cuda/BatchLinearAlgebraLib.h>
+
+#ifdef USE_CUSOLVER
+
+namespace at {
+namespace native {
+
+inline static Tensor column_major_identity_matrix_like(const Tensor& self) {
+  int n = cuda_int_cast(self.size(-2), "self.size(-2)");
+  Tensor self_inv = at::eye({n}, self.options()).expand_as(self).contiguous();
+  self_inv.unsafeGetTensorImpl()->set_stride(self.dim()-2, 1); // These two lines set self_inv to column-major
+  self_inv.unsafeGetTensorImpl()->set_stride(self.dim()-1, n);
+  return self_inv;
+}
+
+template <typename scalar_t>
+inline static void _apply_single_inverse_helper(scalar_t* self_ptr, scalar_t* self_inv_ptr, int* ipiv_ptr, int* info_ptr, int n) {
+  // self_inv_ptr should already be an identity matrix
+
+  auto handle = at::cuda::getCurrentCUDASolverDnHandle();
```

Comment:
I trust that it will correctly set stream?

Reply:
getCurrentCUDASolverDnHandle is basically direct copy of the cublas version with corresponding replacement